src/image.cpp

```diff
@@ -84,6 +84,55 @@ void image::resize(const cv::Mat& input, cv::Mat& output, const cv::Size2i& size
     }
 }
 
+void image::resize_short(const cv::Mat& input, cv::Mat& output, const int target_size)
+{
+    if (target_size == 0)
+    {
+        output = input;
+        return;
+    }
+
+    auto percent = static_cast<float>(target_size) / std::min(input.cols, input.rows);
+
+    auto resized_width  = static_cast<int>(std::round(input.cols * percent));
+    auto resized_height = static_cast<int>(std::round(input.rows * percent));
+    cv::resize(input, output, cv::Size2i(resized_width, resized_height), 0, 0, CV_INTER_LINEAR);
+}
+
+void image::standardize(std::vector<cv::Mat>&      input,
+                        const std::vector<double>& mean,
+                        const std::vector<double>& stddev)
+{
+    if (mean.empty())
+        return;
+    // prep for division
+    cv::Scalar s_mean, s_stddev;
+    for (int i = 0; i < mean.size(); i++)
+    {
+        s_stddev[i] = stddev[i] ? 1. / stddev[i] : 1.;
+        s_mean[i]   = mean[i];
+    }
+
```

Comment:
you can do same thing without loop during init
```cv::Scalar s_mean(mean[0], mean[1], mean[3])```
However I don't understand how it could work. first of all , Scalar always has size =4, so in the loop you try to read 4th element of mean that can cause segmentation fault